crates/driver/src/infra/solver/dto/solution.rs

```diff
@@ -204,6 +223,53 @@ impl Solutions {
             })
             .collect()
     }
+
+    // Function to check that a JIT order signature is signed by the solver
+    fn recover_signer_from_jit_trade_order(
+        jit: &JitTrade,
+        signature: &competition::order::Signature,
+        solver: &Solver,
```

Comment:
Let's pass `eth::DomainSeparator` directly. This can be read from `solver.eth.contracts().settlement_domain_separator()`.